src/main/java/ui/part/WrappingTableCell.java

```diff
@@ -0,0 +1,27 @@
+package ui.part;
+
+import javafx.scene.control.TableCell;
+import javafx.scene.text.Text;
+
+public class WrappingTableCell<S, T> extends TableCell<S, T> {
+
+    @Override
+    public void updateItem(T t, boolean empty) {
+        super.updateItem(t, empty);
+        String item = empty ? "" : String.valueOf(t);
+
+        Text text = new Text(item);
+        text.setStyle(" -fx-text-wrap: true;");
+
+        calculateSize(text);
+        this.widthProperty().addListener((ob, o, n) -> calculateSize(text));
+
+        this.setGraphic(text);
+    }
+
+    private void calculateSize(Text text) {
+        double width = this.getTableColumn().getWidth() - 10;
```

Comment:
why -10? this should be documented.. is it because of the scroll bar? If so, try to get the width of the sroll bar instead of using a magic constant

Reply:
This is only used so the text wrapping is not at the border of the column.
Looks better